src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescriptionJsonConverter.cs

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if NET8_0_OR_GREATER
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Identity.Abstractions
+{
+    /// <summary>
+    /// Custom JSON converter for CredentialDescription to handle proper serialization/deserialization
+    /// of credential information based on SourceType.
+    /// </summary>
+    /// <remarks>BE CAREFUL when you serialize a credential description. The secrets it contains
+    /// will be serialized too depending on the CredentialSource (ClientSecret, Base64Encoded, and Password)</remarks>
+    public class CredentialDescriptionJsonConverter : JsonConverter<CredentialDescription>
+    {
+        /// <inheritdoc/>
+        public override CredentialDescription Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException("Expected start of object");
+            }
+
+            var credentialDescription = new CredentialDescription();
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    return credentialDescription;
+                }
+
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    throw new JsonException("Expected property name");
+                }
+
+                string propertyName = reader.GetString()!;
+                reader.Read();
+
+                switch (propertyName.ToLowerInvariant())
+                {
+                    case "sourcetype":
```

Comment:
could we do 

```cs
case nameof(CredentialDescription.SourceType)
```

And similar for other values?

Reply:
Or not worth it to remove the string literals?

Reply:
Thanks for the suggestion, @keegan-caruso . I did it!